lib/link_expansion/content_cache.rb

```diff
@@ -17,8 +17,20 @@ def find(content_id)
 
   attr_reader :store, :with_drafts, :locale
 
+  def preload_content(editions, content_ids)
+    store = Hash[editions.compact.map { |edition| [edition.content_id, edition] }]
+
+    editions(content_ids).each do |edition|
+      if !store.has_key?(edition.content_id)
+        store[edition.content_id] = edition
+      end
+    end
+
+    store
```

Comment:
I think there's scope to refactor this a bit to make it easier to understand.

method name might make sense more like "build_store" or similar as another `preload_*` gets a bit tricky to grok.

could we do something like:
```
to_preload = content_ids - existing_editions.map(&:content_id)
preloaded_editions = editions(to_preload)
// group them all by content_id to create the store
```

not sure if it only seems more readable to me? could be there's a different way to do it that's easier still - wdyt?